Library/ViewModels/ManageViewPledgeViewModel.swift

```diff
@@ -0,0 +1,66 @@
+import KsApi
+import Prelude
+import ReactiveExtensions
+import ReactiveSwift
+
+public protocol ManageViewPledgeViewModelInputs {
+  func configureWith(_ project: Project, reward: Reward)
+  func viewDidLoad()
+}
+
+public protocol ManageViewPledgeViewModelOutputs {
+  var configurePaymentMethodView: Signal<Project, Never> { get }
+  var configurePledgeSummaryView: Signal<Project, Never> { get }
+  var configureRewardSummaryView: Signal<Reward, Never> { get }
+  var title: Signal<String, Never> { get }
+}
+
+public protocol ManageViewPledgeViewModelType {
+  var inputs: ManageViewPledgeViewModelInputs { get }
+  var outputs: ManageViewPledgeViewModelOutputs { get }
+}
+
+public final class ManageViewPledgeViewModel:
+  ManageViewPledgeViewModelType, ManageViewPledgeViewModelInputs, ManageViewPledgeViewModelOutputs {
+  public init() {
+    let projectAndReward = self.projectAndRewardSignal
```

Comment:
I think we _usually_ use `.combineLatest` here to ensure that both signals have fired at least once.

Reply:
The problem with `.combineLatest` here is that will change the variable's type by combining the signals type. So instead of ` Signal<(Project, Reward), Never>`, it would become ` Signal<((Project, Reward), Void), Never>`. The `.takeWhen` guarantees that the .`projectAndRewardSignal` only emits when `viewDidLoad` after emit without changing the original type. Does it make sense?

Reply:
It does change the signature, but that's not usually an issue because we can do `.map(first)`.

I _think_ problem with using `.takeWhen` is it that it doesn't wait for both signals to fire at least once 🤔 from the description of `.takeWhen`: `"Emits the most recent value of `self` when `other` emits."`.

If the most recent of `self` (ie. `projectAndRewardSignal`) isn't there because that signal hasn't fired once, then it wouldn't work. Most of the time it would probably be fine, but we don't control when `viewDidLoad` gets called because it's a lifecycle function - I'm concerned there could be a race condition where somehow `viewDidLoad` gets called before we can call `configureWith(project:reward)`.

cc @justinswart any feedback/insight on this?

Reply:
Yes, @ifbarrera is correct here in that we would want to use `combineLatest` because that negates the concern about the order in which these signals emit. Using `takeWhen` we must ensure that `configure(with:)` is called first so that it was a value to _take_ when `viewDidLoad` emits. By combining the two we're allowed to call these inputs in any order. Our preference and the pattern we typically use for this is `combineLatest` 👍

Reply:
@Scollaco can you please make sure to make this change in a subsequent PR?